### PR TITLE
[DB-29570] OpenShift provision and E2E test

### PR DESCRIPTION
**Pull request by thebithead**
Shell scripts to provision an OpenShift cluster, and run bash based E2E test script.

Demo/example is in the Jira ticket.

### CODE REVIEW

File: test/bin/openshift_provision.sh [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+# sshKey: |
+#   ssh-rsa obscurecredssssAAAfDAffffffffffffffffHHxpqaG1CRSuTxH8+uk/88ZLZGTMtriXz9cz2Prji4d3lpU3fd38OeaT22lRAddaFcBtW9r6X78i3qRSt8PXabqC7X1MqbPqToeeONDnWvsoVXvyQddddddddddddddddddddddlc9xYO3oWv1j+kcmo9augsUYsqMoXYPOcyj/F6pN2RSfo4YkbMhp1d/352ozAdvssssssssssssssqmgKk2eMxfIvg/683hw3dZfgfffffffffffffffff5NM7z+EPf6TKSUiJggggggggggggggggggggggggggggggSgmxw2sjgwTUsyTYlUTwNxDSV6iq5NgO6UoL tom@thebithead.com
+#
+################################################################################
+# End of example install-config.yaml for provisioning OpenShift on AWS.
+################################################################################
+#
```
Comment (vegichan) [inline test/bin/openshift_provision.sh]: do we need this example?

**Comment by vegichan** [comment]
how do you run this? What are the results? Is it automated?

File: test/bin/openshift_provision.sh [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+tar xvzf openshift-client-linux-${ocp_version}.tar.gz 
+wget https://mirror.openshift.com/pub/openshift-v4/clients/ocp/${ocp_version}/openshift-install-linux-${ocp_version}.tar.gz
+tar -xvzf openshift-install-linux-${ocp_version}.tar.gz
+date
+export PATH="${TEST_DIR}":"${PATH}"
+echo "Creating the OpenShift Cluster"
+openshift-install create cluster
+date
+export KUBECONFIG="${TEST_DIR}/auth/kubeconfig"
+export KUBEADMIN_PASSWORD=$(cat "${TEST_DIR}/auth/kubeadmin-password")
```
Comment (vegichan) [inline test/bin/openshift_provision.sh]: who/what creates this password?

**Comment by thebithead** [comment]
> how do you run this? What are the results? Is it automated?

The E2E test (nuodb_op_system_e2e.sh) does the following:

- Checks the KUBECONFIG
- Displays version and cluster info.
- Creates a nuodb namespace
- Clones the NuoDB Operator GitHub Repo
- Creates the nuodb-operator service account.
- Creates the nuodb-operator role.
- Creates the nuodb-operator role binding.
- Creates the nuodb-op-admin cluster role binding
- Creates secret for registry credentials for Quay.io
- Creates nuodb CRD
- Creates nuodbycsbwl CRD
- Creates nuodbinsightsserver CRD
- Sets required OCP policy
- Deploys the latest NuoDB Operator
- Creates NuoDB test1 database CR: nuodb-operator/test/deploy/crs/nuodb_v2alpha1_nuodb_test1_cr.yaml
- Creates NuoDB Insights Server CR: nuodb-operator/deploy/crds/nuodb_v2alpha1_nuodbinsightsserver_cr.yaml 
- Validates that the SM StatefulSet is READY.
- Validates that the TE Deployment is READY.
- Validates that the Admin StatefulSet is READY.
- Creates NuoDB YCSB Workload CR: nuodb-operator/test/deploy/crs/nuodb_v2alpha1_nuodbycsbwl_test1_cr.yaml
- Validates that Logstash POD is READY.
- Starts the Insights Client POD.
- Runs nuocmd show domain -n nuodb , writing to stdout, and returns zero.
- Runs nuocmd check database --db-name test1, writing to stdout, and returns zero.
- Runs nuocmd check servers --timeout 30, writing to stdout, and returns zero.
- Gets the secret for ElasticSearch.

It validates that command return zero.  Any command that does not succeed with cause the test to exit with a "FAIL" message and the script will exit non-zero.

The script openshift_provsion.sh automates the provisioning and configuration of an OpenShift Cluster that is suitable to run this test.

You can run it from the a unix command line like this:

openshift_provision.sh openshift-install-config.yaml
source ocp_e2e_test_env.sh <full-path-to-openshift-install-dir>
nuodb_op_system_e2e.sh

And then cleanup the test and the cluster like this:

nuodb_op_system_e2e.sh clean
./openshift-install destroy cluster

It is expected that this can be run anywhere, but we expect to run this in Jenkins (when it works)

File: test/bin/openshift_provision.sh (hunk 1 above)
Comment (thebithead) [reply test/bin/openshift_provision.sh]: Yes.  You will need to provide your own config/credentials.

File: test/bin/openshift_provision.sh (hunk 2 above)
Comment (thebithead) [reply test/bin/openshift_provision.sh]: The 'openshift-install create cluster' command does.

**Pull request merged**